packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -16,6 +17,8 @@ import { emptyString } from '../cleanString'
  * ```
  */
 export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
+  let assetResolver: ResolveFn
+
   return {
     name: 'vite:asset-import-meta-url',
     async transform(code, id, options) {
```

Comment:
GitHub doesn't allow me to highlight line41:

```js
const rawUrl = code.slice(urlStart, urlEnd)
```

In development mode if `rawUrl` starts with a `.`, I think we can `continue` the while loop, as we leave the code as is for the browser to resolve like the current Vite behaviour.

Reply:
That makes sense. I was also wondering if it makes sense for us to turn off `preferRelative: true` in the resolver. I decided not to since I believe it's unintuitive for `new URL("./foo.png", import.meta.url)` to have different behavior `new URL("foo.png", import.meta.url)`.

Reply:
Yeah we can leave `preferRelative: true` as is so it's consistent with the glob behaviour.